phpBB/assets/javascript/core.js

```diff
@@ -1640,6 +1640,50 @@ phpbb.lazyLoadAvatars = function loadAvatars() {
 	});
 };
 
+// reCAPTCHA-related variables
+var recaptcha_form = $('.g-recaptcha').parents('form'),
+	submit_button = null,
```

Comment:
sorry, I'm working on the guidelines/standards for 3.3 at the moment (using https://github.com/xojs/xo if you're interested) so I'm going to rip this apart lol...
1. always declare vars using keywords pls do not chain
2. might as well use es2015 syntax `recaptcha_form` should be const and the others should be `let`
3. use arrow functions where possible
4. camelCase vars
5. whats with the console logs? I have not tested the code so if its still WIP then ignore...

for simplicity, you can use this.

```
	const recaptchaForm = $('.g-recaptcha').parents('form');
	let submitButton = null;
	let programaticallySubmitted = false;

	phpbb.recaptchaOnLoad = function () {
		console.log('ahoj');
		// Listen to submit buttons in order to know which one was pressed
		$('input[type="submit"]').each(() => {
			$(this).on('click', () => {
				submitButton = this;
			});
		});

		recaptchaForm.on('submit', e => {
			if (!programaticallySubmitted) {
				grecaptcha.execute();
				e.preventDefault();
			}
		});
	}

	phpbb.recaptchaOnSubmit = function () {
		console.log('submit');
		programaticallySubmitted = true;
		// If concrete button was clicked (e.g. preview instead of submit),
		// let's trigger the same action
		if (submitButton) {
			submitButton.click();
		} else {
			// Rename input[name="submit"] so that we can submit the form
			if (typeof recaptchaForm.submit !== 'function') {
				recaptchaForm.submit.name = 'submit_btn';
			}
			recaptchaForm.submit();
		}
	}
```

Reply:
I didn't know we are allowed to use es2015 already. Cool! 😃